spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/DatastoreOperations.java

```diff
@@ -128,18 +127,12 @@
 	 * Finds objects by using a Cloud Datastore query.
 	 * @param query the query to execute.
 	 * @param entityClass the type of object to retrieve.
-	 * @param <T> the type of object to retrieve.
+	 * @param <T> the type of object to retrieve. If the query was a key-query, then
+	 * {@link Key}s are returned instead of the mapped entities.
 	 * @return a list of the objects found. If no keys could be found the list will be
 	 * empty.
 	 */
-	<T> Iterable<T> query(Query<? extends BaseEntity> query, Class<T> entityClass);
-
-	/**
-	 * Finds Cloud Datastore Keys by using a Cloud Datastore query.
-	 * @param query the query to execute that retrieves only Keys.
-	 * @return the list of keys found.
-	 */
-	Iterable<Key> queryKeys(Query<Key> query);
+	<T> Iterable<?> query(Query query, Class<T> entityClass);
```

Comment:
Since a key query is special and very different from an entity query, can we go back to the two separate method signatures that reflect this and buy us the type-safety?

Reply:
@meltsufin I added back `queryKeys` however I did not decide to use it the part-tree query method. The reason is that i felt it made the code more complicated to read to have that BiFunction there and have it reassigned. The `query` method must support key queries anyway for the GQL query methods, so I think we should still use it in part-tree to simplify the code.

Reply:
Can we put the generics back in though?

Reply:
@meltsufin what do you mean ? The reason they the `<? extends BaseEntity>` was removed was because Key is not an entity.

Reply:
Don't we have `queryKeys` now for that purpose so that we can specialize `query` for entities only.